UpdateUser/handler.ts

```diff
@@ -41,51 +43,64 @@ type IUpdateUserHandler = (
 const getUserFromList = (client: GraphRbacManagementClient, email: string) =>
-  tryCatch(
-    () =>
-      client.users.list({
-        filter: `signInNames/any(x:x/value eq '${email}')`
-      }),
-    toError
-  ).map(userList => userList[0]);
+  pipe(
+    TE.tryCatch(
+      () =>
+        client.users.list({
+          filter: `signInNames/any(x:x/value eq '${email}')`
+        }),
+      toError
+    ),
+    TE.map(userList => userList[0])
+  );
 
 const updateUser = (
   client: GraphRbacManagementClient,
   email: EmailAddress,
   user: User,
   adb2cTokenAttributeName: string,
   userPayload: UserUpdatePayload
-): TaskEither<Error, UserUpdated> =>
-  tryCatch(
-    () =>
-      client.users.update(
-        user.userPrincipalName,
-        withoutUndefinedValues({
-          displayName:
-            userPayload.first_name && userPayload.last_name
-              ? `${userPayload.first_name} ${userPayload.last_name}`
-              : undefined,
-          givenName: userPayload.first_name,
-          surname: userPayload.last_name,
-          // eslint-disable-next-line sort-keys
-          [adb2cTokenAttributeName]: userPayload.token_name
-        })
-      ),
-    toError
-  ).chain(() =>
-    getUserFromList(client, email).chain(userResponse =>
-      fromEither(
-        UserUpdated.decode({
-          email,
-          first_name: userPayload.first_name,
-          id: userResponse.objectId,
-          last_name: userPayload.last_name,
-          token_name: userPayload.token_name
-        }).mapLeft(
-          errs =>
-            new Error(
-              `Error decoding UserUpdated ERROR=${readableReport(errs)}`
+): TE.TaskEither<Error, UserUpdated> =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        client.users.update(
+          user.userPrincipalName,
+          withoutUndefinedValues({
+            displayName:
+              userPayload.first_name && userPayload.last_name
+                ? `${userPayload.first_name} ${userPayload.last_name}`
+                : undefined,
+            givenName: userPayload.first_name,
+            surname: userPayload.last_name,
+            // eslint-disable-next-line sort-keys
+            [adb2cTokenAttributeName]: userPayload.token_name
+          })
+        ),
+      toError
+    ),
+    TE.chain(() =>
+      pipe(
+        getUserFromList(client, email),
+        TE.chain(userResponse =>
+          TE.fromEither(
+            pipe(
+              {
+                email,
+                first_name: userPayload.first_name,
+                id: userResponse.objectId,
+                last_name: userPayload.last_name,
+                token_name: userPayload.token_name
+              },
+              UserUpdated.decode,
+              E.mapLeft(
+                errs =>
+                  new Error(
+                    `Error decoding UserUpdated ERROR=${readableReport(errs)}`
+                  )
+              )
             )
+          )
         )
```

Comment:
```suggestion
        TE.chain(

            pipe(
              {
                email,
                first_name: userPayload.first_name,
                id: userResponse.objectId,
                last_name: userPayload.last_name,
                token_name: userPayload.token_name
              },
              UserUpdated.decode,
              TE.fromEither,
              TE.mapLeft(
                errs =>
                  new Error(
                    `Error decoding UserUpdated ERROR=${readableReport(errs)}`
                  )
              )
            )
        )
```

Reply:
You would need `userResponse =>` anyway

Reply:
fixed in https://github.com/pagopa/io-functions-admin/pull/165/commits/409be2e4d1eaf1271b27f010aa2cacd12e7ce248